bundle/edu.gemini.catalog/src/main/scala/edu/gemini/catalog/api/MagnitudeConstraints.scala

```diff
@@ -45,36 +47,57 @@ object SaturationConstraint {
     Order.orderBy(_.brightness)
 }
 
+sealed trait MagnitudeFilter {
+  def filter(t: SiderealTarget): Boolean
+
+  def referenceBand: MagnitudeBand
+  def faintnessConstraint: FaintnessConstraint
+  def saturationConstraint: Option[SaturationConstraint]
+}
+
 /**
- * Describes constraints for the magnitude of a target
- * See OT-19.
+ * Defines a typeclass of classes that can adjust MagnitudeConstraints, e.g. given certain conditions
  */
-case class MagnitudeConstraints(band: MagnitudeBand, faintnessConstraint: FaintnessConstraint, saturationConstraint: Option[SaturationConstraint]) {
+trait ConstraintsAdjuster[T] {
+  def adjust(t: T, mc: MagnitudeConstraints): MagnitudeConstraints
+}
 
-  def this(faint:Magnitude) = this(faint.band, FaintnessConstraint(faint.value), None)
+/**
+ * Defines a class that can extract a magnitude from a target
+ */
+sealed trait MagnitudeExtractor {
+  def extract(t: SiderealTarget):Option[Magnitude]
+  def bandSupported(b: MagnitudeBand):Boolean
+}
 
-  /**
-   * Maps a transformation into a new MagnitudeConstraints
-   */
-  def map(f: Magnitude => Magnitude): MagnitudeConstraints = {
-    val mappedFaintness = f(faintnessConstraint.toMagnitude(band)) 
-    val mappedSaturation = saturationConstraint.map(_.toMagnitude(band)).map(f)
+/**
+ * Search for a specific band on the target
+ */
+case class SingleBandExtractor(band: MagnitudeBand) extends MagnitudeExtractor {
+  override def extract(t: SiderealTarget) = t.magnitudeIn(band)
+  override def bandSupported(b: MagnitudeBand) = b === band
+}
 
-    val fl = FaintnessConstraint(mappedFaintness.value)
-    val sl = mappedSaturation.map(s => SaturationConstraint(s.value))
+/**
+ * Finds the first on a list of bands
+ */
+case class FirstBandExtractor(bands: NonEmptyList[MagnitudeBand]) extends MagnitudeExtractor {
+  override def extract(t: SiderealTarget) = bands.map(t.magnitudeIn).list.find(_.isDefined).flatten
+  override def bandSupported(b: MagnitudeBand) = bands.list.contains(b)
+}
```

Comment:
Right, agreed.  I just want it to be clear that you are querying for and getting results based on one of r', R, or UC.  I think if you explicitly query for R the results should be relative to R and nothing else, just as for H or other bands. So as you say, something that explicitly represents the concept of "one of these bands, in this order of preference".